Module-5/index.js

```diff
@@ -0,0 +1,74 @@
+import url from 'url';
+import http from 'http';
+import { handleMalformedRequest } from './helpers/helpers.js';
+import { sendHobbies, sendUserInfo, sendUsersList } from './helpers/GETHandlers.js';
+import { handleCreateUser } from './helpers/POSTHandlers.js';
+import { handleUpdateHobbies, handleUpdateUser } from './helpers/PATCHHandlers.js';
+import { handleDeleteUser } from './helpers/DELETEHandlers.js';
+
+
+const server = http.createServer((req, res) => {
+  res.setHeader('Content-Type', 'application/json');
+  const regex = /(users)(?:\/(\d+))?(\/hobbies)?/;
+  const parsedUrl = url.parse(req.url);
+  const path = regex.exec(parsedUrl.pathname);
+
+  if (path === null) {
+    handleMalformedRequest(res, { code: 500, message: 'Invalid request' });
+  }
+
+  const [ , root, userId, hobbies ] = path;
+
+  switch (req.method) {
+    case 'GET': {
+      if (root === 'users' && userId && hobbies) {
```

Comment:
I would probably flip the checks or methods and roots – instead of going from the method, you could make your way from request root:

```suggestion
  switch (root) {
    case 'users': {
      if (req.method === GET_METHOD && userId && hobbies) {
        ...
      }
```
or better yet – you could nest `switch..case` statements
although, in a long run it may not be the perfect solution for handling complex endpoints, but for this small application you will not feel the performance degradation